pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1866,3 +1846,33 @@ func newObjectMeta(name, email string) metav1.ObjectMeta {
 		},
 	}
 }
+
+func TestTransformUsername(t *testing.T) {
+	assertName(t, "some", "some@email.com")
+	assertName(t, "so-me", "so-me@email.com")
+	assertName(t, "some", "some")
+	assertName(t, "so-me", "so-me")
```

Comment:
```suggestion
	assertName(t, "so-me", "so-me")
	assertName(t, "so-me", "so-----me")
```

Reply:
this is already indirectly covered by `assertName(t, "so-me", "so:#$%!$%^&me")` but it's good to add also the direct test. Thanks :+1: